org.lflang/src/org/lflang/generator/cpp/CppInstanceGenerator.kt

```diff
@@ -36,29 +36,49 @@ class CppInstanceGenerator(
     private val fileConfig: CppFileConfig,
     private val errorReporter: ErrorReporter
 ) {
-    private val Instantiation.isEnclave: Boolean get() = AttributeUtils.isEnclave(this)
+    companion object {
+        val Instantiation.isEnclave: Boolean get() = AttributeUtils.isEnclave(this)
 
-    private val Instantiation.hasEachParameter: Boolean
-        get() = AttributeUtils.getBooleanAttributeParameter(
-            AttributeUtils.getEnclaveAttribute(this), AttributeSpec.EACH_ATTR
-        ) ?: false
+        val Instantiation.hasEachParameter: Boolean
+            get() = AttributeUtils.getBooleanAttributeParameter(
+                AttributeUtils.getEnclaveAttribute(this), AttributeSpec.EACH_ATTR
+            ) ?: false
 
-    val Instantiation.cppType: String
-        get() {
-            return if (reactor.isGeneric)
+        private val Instantiation.reactorType: String
+            get() = if (reactor.isGeneric)
                 """${reactor.name}<${typeArgs.joinToString(", ") { it.toText() }}>"""
             else
                 reactor.name
-        }
+
+        val Instantiation.cppClass: String
+            get() = if (isEnclave) enclaveWrapperClassName else reactorType
+
+        val Instantiation.enclaveWrapperClassName get() = "EnclaveWrapper_$name"
+    }
+
+    private fun Instantiation.generateWrapper(): String = """
+            |struct $enclaveWrapperClassName {
+            |  ${if (!hasEachParameter) "inline static " else ""}std::unique_ptr<reactor::Environment> __lf_env{nullptr};
+            |  std::unique_ptr<$reactorType> __lf_instance{nullptr};
+            |  
+            |  $enclaveWrapperClassName(const std::string& name, reactor::Reactor* container, $reactorType::Parameters&& params) {
+            |    if (__lf_env == nullptr) {
```

Comment:
We use the `__lf_` prefix for some variables to avoid name clashes. It is unlikely, but if someone would name their state variable or some other reactor component `lf_env_`, we would see compile errors. `__` is kind of a protected namespace in C/C++ that is not supposed to be used by users.